tests/integration/providers/nfsProvider.spec.ts

```diff
@@ -0,0 +1,63 @@
+import jsLogger from '@map-colonies/js-logger';
+import { container } from 'tsyringe';
+import { randFileExt, randWord } from '@ngneat/falso';
+import httpStatus from 'http-status-codes';
+import config from 'config';
+import { getApp } from '../../../src/app';
+import { SERVICES } from '../../../src/common/constants';
+import { NFSConfig } from '../../../src/common/interfaces';
+import { NFSHelper } from '../../helpers/nfsHelper';
+import { AppError } from '../../../src/common/appError';
+import { NFSProvider } from '../../../src/providers/nfsProvider';
+
+describe('NFSProvider', () => {
+  let provider: NFSProvider;
+  const nfsConfig = config.get<NFSConfig>('NFS');
+  let nfsHelper: NFSHelper;
+
+  beforeAll(() => {
+    getApp({
+      override: [
+        { token: SERVICES.PROVIDER_CONFIG, provider: { useValue: nfsConfig } },
+        { token: SERVICES.LOGGER, provider: { useValue: jsLogger({ enabled: false }) } },
+      ],
+    });
+    provider = container.resolve(NFSProvider);
+    nfsHelper = new NFSHelper(nfsConfig);
+  });
+
+  beforeEach(() => {
+    nfsHelper.initNFS();
+  });
+
+  afterEach(async () => {
+    await nfsHelper.cleanNFS();
+    jest.clearAllMocks();
+  });
+
+  describe('deleteFile', () => {
+    it('Should delete an existing file from NFS', async () => {
+      const model = randWord();
+      const file = `${randWord()}.${randFileExt()}`;
+      await nfsHelper.createFileOfModel(model, file);
+      const filePath = `${model}/${file}`;
+
+      const result = await provider.deleteFile(filePath);
+      const fileExists = await nfsHelper.fileExists(filePath);
+
+      expect(result).toBeUndefined();
+      expect(fileExists).toBe(false);
+    });
+
+    it('Should proccess deleting of non-existing file in NFS', async () => {
```

Comment:
Same as handle. The test should be simple and clear. From just seeing the title, you should understand what are you testing.
It is not clear what you are testing. What do you understand from "Should process (not proccess) deleting"? In the previous test you wrote simple and clear that the function should delete. There is no confusion about that.
"Should process deleting" can be anything. Should the function delete? throw error? You have to be clear and simple in your code.

Reply:
ok

Reply:
It's still a wrong sentence. Think what you are trying to test

Reply:
ok